File: abc/abc153/abc153_b.cs
// https://atcoder.jp/contests/abc153/submissions/9740398

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        static void Main()
        {
            var input = Console.ReadLine().Split().Select(int.Parse).ToArray();
            var H = input[0];
            var N = input[1];

            var A = Console.ReadLine().Split().Select(int.Parse).ToArray();

            for (int i = 0; i < N; i++)
            {
                H -= A[i];
            }
            Console.WriteLine(H <= 0 ? "Yes" : "No");
        }
    }
}

File: abc/abc153/abc153_b.fs
// https://atcoder.jp/contests/abc153/submissions/11914877

let [| H; N |] = stdin.ReadLine().Split() |> Array.map int

let As =
    stdin.ReadLine().Split()
    |> Array.map int
    |> Array.sum

if H - As <= 0 then "Yes"
else "No"
|> stdout.WriteLine
